### PR TITLE
feat: add apm.addMetadataFilter(fn)

### PR DESCRIPTION
**Pull request by trentm**
Fixes: #1916

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update TypeScript typings
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

Note: still draft until the http-client PR (https://github.com/elastic/apm-nodejs-http-client/pull/154) and package.json is updated.

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -279,6 +279,7 @@ class Config {
           globalLabels: maybePairsToObject(conf.globalLabels),
           hostname: conf.hostname,
           environment: conf.environment,
+          metadataFilters: agent._metadataFilters,
```
Comment (astorm) [inline lib/config.js:282]: I don't see where/how this is used.  There's no `agent._metadataFilters` property assigned anywhere else -- `addMetadataFilter` adds its functions directly to the transport/client object.  Old code, or something else?

**Review by astorm — COMMENTED** [review]
Overall looks good.  The new API method does what it says it does and has reasonable guards in place.  The new docs are clear to me, and the caveats added to the other filter functions make sense.  

One question about a lingering object property before approving, but otherwise 👍

File: lib/config.js (hunk 1 above)
Comment (trentm) [reply lib/config.js]: Good catch. Cruft left over from an earlier impl. Dropped in commit 26b897a.

**Review by astorm — APPROVED** [review]
👍

**Pull request merged**